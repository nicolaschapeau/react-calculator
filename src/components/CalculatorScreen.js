import React from 'react'


// Component
export default class CalculatorScreen extends React.Component {
    // Render
    render () {
        return (
            <div className="calculator-displayer">
                <p 
                    className="calculator-expression"
                >
                    {this.props.expression}
                </p>
                <p 
                    className="calculator-value"
                >
                    {this.props.value}
                </p>
            </div>
        )
    }
}