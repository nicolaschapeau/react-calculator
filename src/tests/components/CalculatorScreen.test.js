import React from 'react';
import { shallow } from 'enzyme';
import CalculatorScreen from '../../components/CalculatorScreen';
import toJSON from 'enzyme-to-json'



test('Should render calculator screen', () => {
    const wrapper = shallow(<CalculatorScreen expression="1+2" value="3" />)

    expect(toJSON(wrapper)).toMatchSnapshot()
})


test('Should render calculator screen with good expression', () => {
    const wrapper = shallow(<CalculatorScreen expression="1+2" value="3" />)

    expect(wrapper.find('p.calculator-expression').text()).toBe('1+2')
})


test('Should render calculator screen with good value', () => {
    const wrapper = shallow(<CalculatorScreen expression="1+2" value="3" />)

    expect(wrapper.find('p.calculator-value').text()).toBe('3')
})
