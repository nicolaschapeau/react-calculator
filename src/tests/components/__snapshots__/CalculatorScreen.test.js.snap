// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render calculator screen 1`] = `
<div
  className="calculator-displayer"
>
  <p
    className="calculator-expression"
  >
    1+2
  </p>
  <p
    className="calculator-value"
  >
    3
  </p>
</div>
`;
